lexers/LexTOML.cxx

```diff
@@ -263,8 +263,7 @@ void ColouriseTOMLDoc(Sci_PositionU startPos, Sci_Position lengthDoc, int initSt
 							keyState = TOMLKeyState::End;
 							sc.SetState(SCE_TOML_DEFAULT);
 						} else if (chNext != '.' && chPrevNonWhite != '.') {
-							sc.ChangeState(SCE_TOML_ERROR);
```

Comment:
Since both branch looks identical, the whole if can be simplified as:
```cpp
} else if (sc.state == SCE_TOML_KEY && !IsTOMLUnquotedKey(sc.ch)) {
	keyState = TOMLKeyState::End;
	sc.SetState(SCE_TOML_DEFAULT);
}
```

Reply:
Thanks for looking at it - yes, you are right, the branches can be merged, but the conditions checking `=` and `.` have to stay. I've just re-pushed the patch with
```C
} else if (sc.state == SCE_TOML_KEY && !IsTOMLUnquotedKey(sc.ch)) {
	const int chNext = GetLineNextChar(sc);
	if (chNext == '=' || (chNext != '.' && chPrevNonWhite != '.')) {
		keyState = TOMLKeyState::End;
		sc.SetState(SCE_TOML_DEFAULT);
	}
}
```